### PR TITLE
Bug 1216556 - Improve camera lazy loading to reduce startup time.

### PR DESCRIPTION
**Pull request by aosmond**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2ee4c5b1ba21835e88104fa825b9e439b2e4bab9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=471e3c1e356b179add33650ee9bf0355ce562056

File: apps/camera/js/lib/geo-location.js [hunk 1]
```diff
@@ -39,7 +33,7 @@ function GeoLocation() {
  */
 GeoLocation.prototype.watch = function() {
   if (!this.watcher) {
-    this.watcher = geolocation.watchPosition(this.setPosition);
+    this.watcher = navigator.geolocation.watchPosition(this.setPosition);
```
Comment (aosmond) [inline apps/camera/js/lib/geo-location.js]: Note that the mere act of accessing navigator.geolocation creates the geolocation object along with underlying services, etc. So getting rid of the local above took that creation off the critical path.

File: apps/camera/js/app.js [hunk 2]
```diff
@@ -261,16 +264,30 @@ App.prototype.onCriticalPathDone = function() {
...
+/**
+ * Lazy load activity dependencies.
+ *
+ * @private
+ */
+App.prototype.onActivity = function() {
+  if (!this.criticalPathDone) {
+    this.controllers.lazy.push('controllers/confirm');
+  } else {
+    this.loadLazyControllers(['controllers/confirm']);
```
Comment (aosmond) [inline apps/camera/js/app.js]: As far as I can tell, this code path is never reached, not with a reused camera process, nor with a fresh one. What seems to happen is the existing camera process will restart itself entirely if already opened when a pick from MMS is required.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=db053b4e448de7c7513738fb16360ce68c8cf60e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=737b7f37be65ffb331f50502cb793350df4866ad

**Pull request merged**